src/main/java/Deadline.java

```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task{
+
+    private String by;
+
+    public Deadline(String task, String by) {
+        super(task);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        if (isDone) {
+            return "[D]" + super.toString() + "(by:" + by + ")";
+        } else {
+            return "[D]" + super.toString() + "(by:" + by + ")";
+        }
+    }
```

Comment:
Perhaps you could make this more concise. There are some of this instance in later part of the code.
```suggestion
    public String toString() {
        return "[D]" + super.toString() + "(by:" + by + ")";
    }
```